pkg/network/setup/netpod/discover.go

```diff
@@ -134,6 +142,123 @@ func (n NetPod) storePodInterfaceData(vmiSpecIface v1.Interface, ifaceState nmst
 	return nil
 }
 
+func (n NetPod) storeBridgeBindingDHCPInterfaceData(currentStatus *nmstate.Status, podIfaceStatus nmstate.Interface, vmiSpecIface v1.Interface, podIfaceName string) error {
+	var dhcpConfig cache.DHCPConfig
+	dhcpConfig.IPAMDisabled = true
+	if ipAddress := firstIPGlobalUnicast(podIfaceStatus.IPv4); ipAddress != nil {
+		dhcpConfig.IPAMDisabled = false
+
+		addr, iperr := vishnetlink.ParseAddr(fmt.Sprintf("%s/%d", ipAddress.IP, ipAddress.PrefixLen))
+		if iperr != nil {
+			return iperr
+		}
+		dhcpConfig.IP = *addr
+
+		mac, err := resolveMacAddress(podIfaceStatus.MacAddress, vmiSpecIface.MacAddress)
+		if err != nil {
+			return err
+		}
+		dhcpConfig.MAC = mac
+
+		linkRoutes, err := filterIPv4RoutesByInterface(currentStatus, podIfaceName)
+		if err != nil {
+			return err
+		}
+		dhcpConfig.Gateway = net.ParseIP(linkRoutes[0].NextHopAddress)
+
+		otherRoutes, err := filterRoutesByNonLocalDestination(linkRoutes, addr)
+		if err != nil {
+			return err
+		}
+
+		dhcpRoutes, err := filterRoutesToDelegateThroughDHCP(otherRoutes)
+		if err != nil {
+			return err
+		}
+		if len(dhcpRoutes) > 0 {
+			dhcpConfig.Routes = &dhcpRoutes
+		}
+	}
+
+	log.Log.V(4).Infof("The generated dhcpConfig: %s\nRoutes: %+v", dhcpConfig.String(), dhcpConfig.Routes)
+	if err := cache.WriteDHCPInterfaceCache(n.cacheCreator, strconv.Itoa(n.podPID), podIfaceName, &dhcpConfig); err != nil {
+		return fmt.Errorf("failed to save DHCP configuration: %w", err)
+	}
+
+	return nil
+}
+
+func filterRoutesToDelegateThroughDHCP(otherRoutes []nmstate.Route) ([]vishnetlink.Route, error) {
+	var dhcpRoutes []vishnetlink.Route
+	for _, nmstateRoute := range otherRoutes {
+		isDefaultRoute := nmstateRoute.Destination == nmstate.DefaultDestinationRoute(vishnetlink.FAMILY_V4).String()
+		var dstAddr *net.IPNet
+		if !isDefaultRoute {
+			_, ipNet, perr := net.ParseCIDR(nmstateRoute.Destination)
+			if perr != nil {
+				return nil, perr
+			}
+			dstAddr = ipNet
+		}
+		route := vishnetlink.Route{
+			Dst: dstAddr,
+			Gw:  net.ParseIP(nmstateRoute.NextHopAddress),
+		}
+		dhcpRoutes = append(dhcpRoutes, route)
+	}
+	return dhcpRoutes, nil
+}
+
+// filterRoutesByNonLocalDestination filters out local routes (the destination is of the local link).
+// Default routes should not be filter out.
+func filterRoutesByNonLocalDestination(linkRoutes []nmstate.Route, addr *vishnetlink.Addr) ([]nmstate.Route, error) {
+	var otherRoutes []nmstate.Route
+	for _, route := range linkRoutes {
+		_, dstIPNet, perr := net.ParseCIDR(route.Destination)
+		if perr != nil {
+			return nil, perr
+		}
+		isDefaultRoute := route.Destination == nmstate.DefaultDestinationRoute(vishnetlink.FAMILY_V4).String()
+		localDestination := dstIPNet.Contains(addr.IP)
+		if isDefaultRoute || !localDestination {
```

Comment:
Why is it required to check `isDefaultRoute`. Can the default route have a local destination?
Also, the original code checked the source is not local while here the destination is verified. Why?
```
route.Src != nil && route.Src.Equal(nic.IP.IP)
```